tools/codegen/api/types.py

```diff
@@ -266,11 +266,16 @@ def no_default(self) -> 'Binding':
             argument=self.argument,
         )
 
-    def decl(self) -> str:
+    def decl(self, *, func_ptr_cast: bool = False) -> str:
         mb_default = ""
         if self.default is not None:
             mb_default = f"={self.default}"
-        return f"{self.type} {self.name}{mb_default}"
+
+        # casting only needs to know the type
+        if func_ptr_cast:
```

Comment:
sigh... we actually only need this for CPU fallbacks, so I'll add a note to kill it when we make a boxed kernel.